core/common/src/main/java/alluxio/wire/BlockInfo.java

```diff
@@ -23,6 +25,7 @@
 /**
  * The block descriptor.
  */
+@PublicApi
```

Comment:
I would prefer to add that as a TODO if that's okay with you. The advantage of the wrapper would be it that it allows us to change the internal type in a backwards incompatible manner and also to hide the block ids, worker ids, and tier aliases from the user. However, I am not sure it is worth the effort and I am also not sure where to put these wrappers.

Since other wire types are already part of the public API, I suggest we add these as well and if need be, address removing wire types from the public API in a separate PR.